src/task.rs

```diff
@@ -7,12 +56,70 @@ use std::sync::Arc;
 use std::task::{Context, Poll};
 use std::time::{Duration, Instant};
 
-pub struct TaskMetrics {
-    metrics: Arc<Metrics>,
+/// Monitors key metrics of instrumented tasks.
+/// 
+/// ### Usage
+/// A [`Monitor`] tracks key [metrics][Metrics] of [`Future`]s that have been [instrumented][`Monitor::instrument`]
+/// with the monitor. 
+/// 
+/// In the below example, a [`Monitor`] is [constructed][Monitor::new] and used to 
+/// [instrument][Monitor::instrument] three worker tasks; meanwhile, a fourth task
+/// prints [metrics][Metrics] in 500ms [intervals][Monitor::intervals].
+/// ```
+/// use std::time::Duration;
+/// use tokio_metrics::task;
+/// 
+/// #[tokio::main]
+/// async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
+///     // construct a metrics monitor
+///     let metrics_monitor = task::Monitor::new();
+/// 
+///     // print task metrics every 500ms
+///     {
+///         let metrics_monitor = metrics_monitor.clone();
+///         tokio::spawn(async move {
+///             for interval in metrics_monitor.intervals() {
+///                 // pretty-print the metric interval
+///                 println!("{:?}", interval);
+///                 // wait 500ms
+///                 tokio::time::sleep(Duration::from_millis(500)).await;
+///             }
+///         });
+///     }
+/// 
+///     // instrument some tasks and await them
+///     // note that the same monitor can be used for multiple tasks
+///     tokio::join![
+///         metrics_monitor.instrument(do_work()),
+///         metrics_monitor.instrument(do_work()),
+///         metrics_monitor.instrument(do_work())
+///     ];
+/// 
+///     Ok(())
+/// }
+/// 
+/// async fn do_work() {
+///     for _ in 0..25 {
+///         tokio::task::yield_now().await;
+///         tokio::time::sleep(Duration::from_millis(100)).await;
+///     }
+/// }
+/// ```
+/// 
+/// ### Limitations
+/// Internally, this crate uses [`u64`]s to represent event counts and durations (measured in nanoseconds).
+/// Consequently, [`Metrics::num_fast_polls`], for example, can only count up to 2<sup>64</sup> - 1 polls,
+/// and  [`Metrics::total_time_slow_poll`] can only represent durations up to 584 years. These limits are
+/// not reachable in practice.
+/// 
+#[derive(Clone)]
+pub struct Monitor {
+    metrics: Arc<RawMetrics>,
 }
 
 pin_project! {
-    pub struct InstrumentedTask<T> {
+    /// A future that has been instrumented with [`Monitor::instrument`].
```

Comment:
Take it or leave it, but instead of `future`, maybe `async task`. "Future" w/ async rust is kind of tricky as it is being pushed down to the level of "implementation detail". The Tokio docs tries to avoid futures until around the end.